x/subscription/README.md

```diff
@@ -0,0 +1,202 @@
+```
+
+This will create the `FutureSubscription` object inside the `Subscription` object, as can be seen above.
+The new subscription will be triggered once the current active subscription expires.
+
+Users can decide on the duration of this future subscription, and they'll pay the full amount for the entire period immediately.
+If the plan changes during that period, users who bought this plan before it changed using the `--advance-purchase` flag, they won't be affected by that change.
+
+If a user tries to replace the future subscription with another, the new plan's price must be higher, considering the amount of days bought.
+Meaning, if user originally bough X days of a plan with price A, and now wants to advance purchase Y days of a different plan with price B, than the following must be suffice:
+
+$$
+Y * B > X * A
+$$
+
+### Auto Renewal
+
+Users can decide to if they want the subscription to auto-renew itself every month, using the subscription buy transaction command:
+
+```bash
+lavad tx subscription auto-renewal [true/false] [optional: plan-index] [optional: consumer] [flags]
+```
+
+Here, users can decide to renew their subscription to any plan index, wether it's cheaper or more expensive than the current active subscription's plan.
+
+The tokens will be taken from the user's account only when at the end of each month, prior to renewing the subscription, and for one month only.  
+If the user does not have sufficient funds for the renewal, the subscription will expire.
+
+If a user has a future subscription configured and auto-renewal is also set, then auto-renewal will take effect only after the future subscription expires.
+
+## Parameters
+
+The subscription module does not contain parameters.
+
+## Queries
+
+The subscription module supports the following queries:
+
+| Query                  | Arguments             | What it does                                                   |
+| ---------------------- | --------------------- | -------------------------------------------------------------- |
+| `current`              | consumer (string)     | Shows the current subscription of a consumer to a service plan |
+| `list`                 | subscription (string) | Shows all current subscriptions                                |
+| `list-projects`        | none                  | Shows all the subscription's projects                          |
+| `next-to-month-expiry` | none                  | Shows the subscriptions with the closest month expiry          |
+| `params`               | none                  | Shows the parameters of the module                             |
+
+## Transactions
+
+All the transactions below require setting the `--from` flag and gas related flags.
+
+The subscription module supports the following transactions:
+
+| Transaction    | Arguments                                                                               | What it does                                  |
```

Comment:
in the "what it does" section, specify when the TX will be applied (immediately? next epoch?)

Reply:
Added in c26d2c9b0f58a4e7d315762a912e39e616bee8e1